front/components/product/product.html

```diff
@@ -19,7 +19,7 @@
       <b>qty:</b><input type="text" class="qty" ng-model="qty"><br>
 
       <div class="text-center">
-        <input type="submit" value="add to cart">
```

Comment:
Here, you probably want a button.  And you do not need to pass the function any variables because these will be accessible by doing

$scope.product

and 

$scope.qty